src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -7011,6 +7026,19 @@ instruct castLL(rRegL dst)
   ins_pipe(empty);
 %}
 
+instruct castLL_checked(rRegL dst, rRegL tmp, rFlagsReg cr)
```

Comment:
Optimization idea: considering the range is statically known, `tmp` is not needed when range boundaries fit into signed int. Worth adding an extra AD instruction to benefit from that.